platforms/core-configuration/file-collections/src/main/java/org/gradle/api/internal/file/AbstractFileTreeElement.java

```diff
@@ -106,7 +107,13 @@ private void copyFile(File target) throws IOException {
     }
 
     @Override
+    @Deprecated
     public int getMode() {
+        DeprecationLogger.deprecateMethod(FileTreeElement.class, "getMode()")
+            .replaceWith("getPermissions()")
```

Comment:
🤔  Can't we suggest a replacement that will just work? What about `getPermissions().toUnixNumeric()` in this case? There are other cases like this in the PR.